src/c++/perf_analyzer/genai-perf/genai_perf/llm_metrics.py

```diff
@@ -639,10 +639,15 @@ def _extract_openai_text_output(self, response: str) -> str:
         completions = data["choices"][0]
 
         text_output = ""
-        if data["object"] == "text_completion":  # legacy
+        if "object" not in data:
+            # FIXME: TPA-47 workaround for vLLM not following OpenAI Completions
+            # API specification when streaming, missing 'object' field:
+            # https://platform.openai.com/docs/api-reference/completions
+            text_output = completions.get("text", "")
+        elif data["object"] == "text_completion":  # legacy
             text_output = completions.get("text", "")
         elif data["object"] == "chat.completion":  # non-streaming
-            text_output = completions["message"]["content"]
+            text_output = completions["message"].get("content", "")
```

Comment:
If this get does not find the entry it is looking for, does this break our metrics later?

Reply:
I have no idea, I was just trying to make it follow the pattern elsewhere of providing an empty string if that field is missing. Do you think it's better to error there? Should all other cases fail if a field is missing?

Reply:
Nah. 
For now, i think we can go with this.